nlp/2021_03_18_pke_keyword_extraction/pke_keyword_extraction.ipynb

```diff
@@ -0,0 +1,503 @@
+      "metadata": {
+        "id": "rvXj0fPGMXX9"
+      },
+      "source": [
+        "For a long time the go-to library for graph-based keyword extraction has been Gensim. However, the next major release—version 4.0 which is currently in beta—will [remove the entire summarisation module](https://github.com/RaRe-Technologies/gensim/wiki/Migrating-from-Gensim-3.x-to-4#12-removed-gensimsummarization), which includes the keyword extraction functionality.\n",
+        "\n",
+        "This motivates an update on **keyword extraction** from a previous Tip of the week on extractive summarization which included a section about keyword extraction with Gensim."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eQN6D4QVB2vM"
+      },
+      "source": [
+        "This notebook gives a brief overview of **pke** an **open-source keyphrase extraction toolkit**, that is easy to use, provides a wide range of keyword extraction methods which makes it easy to benchmark different approaches in order to choose the right algorithm for the problem at hand.\n",
+        "\n",
+        "In contrast to Gensim, which only provided keyword extraction using the widely known TextRank algorithm, `pke` offers [statistical models](https://boudinfl.github.io/pke/build/html/unsupervised.html#statistical-models), a [variety of TextRank flavours](https://boudinfl.github.io/pke/build/html/unsupervised.html#graph-based-models), as well as [simple supervised methods](https://boudinfl.github.io/pke/build/html/supervised.html) which even come pre-trained.\n",
+        "\n",
+        "In `pke` **preprocessing is built-in** using **spaCy**. This means the times of tormenting non-English languages with numerous preprocessing steps in order to make it look like English are finally over. *Bon vent !*\n",
+        "\n",
+        "Follow the code in this notebook to see how to use `pke` to extract keywords and a camparison of the extracted keywords.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "A_vPjpflJdkF"
+      },
+      "source": [
+        "---\n",
+        "\n",
+        "*My sincere apology to everyone who had to read this headline!*"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "E5fga3OkHom-"
+      },
+      "source": [
+        "## 🏗 Getting started: Install packages & download models\n",
+        "\n",
+        "The below cells will setup everything that is required to get started with keyword extraction:\n",
+        "\n",
+        "* Install packages\n",
+        "* Download additional resources"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "lbWP_PcNsL8Q"
+      },
+      "source": [
+        "# Install pke\n",
+        "!pip install --quiet git+https://github.com/boudinfl/pke.git\n",
+        "\n",
+        "# Download additional resources\n",
+        "!python -m nltk.downloader stopwords\n",
```

Comment:
In your experience, does using stopwords give a good boost in performance? Because I'm personally not the biggest fan of nltk

Reply:
Some of the extraction methods take a list of stop words as input. I followed suit, since it's the default behavior and uses NLTK as well which provides stop word lists for multiple languages.

Reply:
Gotcha. If we can avoid NLTK that's great, but if it makes sense in this case: 👌